File: Minimum indexed character - GFG/minimum-indexed-character.cpp
//{ Driver Code Starts
//Initial template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++

class Solution
{
  public:
    //Function to find the minimum indexed character.
    int minIndexChar(string str, string patt)
    {
        int hs[256];
        memset(hs,-1,sizeof(hs));
        for(int i=str.size()-1;i>=0;i--){
            hs[str[i]]=i;
        }
        int n = patt.size();
        int ans = str.size();
        for(int i=0;i<n;i++){
            if(hs[patt[i]]>=0){
                ans=min(ans,hs[patt[i]]);
            }
        }
        if(ans == str.size()){
            ans = -1;
        }
        return ans;
    }
};

//{ Driver Code Starts.

int main()
{
	int t;
    cin>>t;
    while(t--)
    {
        string str;
        string patt;
        cin>>str;
        cin>>patt;
        Solution obj;
        cout<<obj.minIndexChar(str, patt)<<endl;
    }
	return 0;
}

// } Driver Code Ends

File: Minimum indexed character - GFG/minimum-indexed-character.java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;


// } Driver Code Ends
//User function template for JAVA

class Solution
{
    //Function to find the minimum indexed character.
    public static int minIndexChar(String str, String patt)
    {
        int[] hs = new int[256];
        for(int i=0;i<256;i++){
            hs[i]=-1;
        }
        int n = patt.length();
        int m = str.length();
        for(int i=0;i<n;i++){
            hs[patt.charAt(i)]++;
        }
        for(int i=0;i<m;i++){
            if(hs[str.charAt(i)]>=0){
                return i;
            }
        }
        return -1;
    }
}

//{ Driver Code Starts.

class GFG {
	public static void main (String[] args) 
	{
		Scanner sc=new Scanner(System.in);
		int t=sc.nextInt();
		while(t>0)
		{
		    t--;
		    
		    String s1=sc.next();
		    String s2=sc.next();
		    
		    int res = new Solution().minIndexChar(s1, s2);
		    System.out.println(res);
		}
	}
}

// } Driver Code Ends